src/tests/TestXMLTools/TestXMLTools.cpp

```diff
@@ -0,0 +1,223 @@
+	QXmlStreamReader reader(input);
+	QBuffer outBuf;
+	QXmlStreamWriter writerOut(&outBuf);
+	outBuf.open(QIODevice::WriteOnly);
+	int paragraphs = 0;
+	XMLTools::recurseParse(reader, writerOut, paragraphs);
+	outBuf.close();
+	return QString(outBuf.data());
+}
+
+void TestXMLTools::testParseExtractBody() {
+	QCOMPARE(doParse(QLatin1String("<html><head></head><body>one</body></html>")), QLatin1String("one"));
+	QCOMPARE(doParse(QLatin1String("<!DOCTYPE invalid><html><head></head><body>one</body></html>")), QLatin1String("one"));
+	QCOMPARE(doParse(QLatin1String("<html><head><title>ttt</title></head><body>one</body></html>")), QLatin1String("one"));
+	QCOMPARE(doParse(QLatin1String("<html><head></head><body><b>one</b></body></html>")), QLatin1String("<b>one</b>"));
+}
+
+void TestXMLTools::testParseConvertStyleBold() {
+	// The numeric values for font-weight provide a more granular option to specify boldness.
+	// On fonts with two boldness levels (normal and bold) this is fine (100-500, 600-900).
+	// For fonts wiht more boldness levels, the conversion we do here from a weight value to a <b> tag is problematic.
+	// http://doc.qt.io/qt-4.8/stylesheet-reference.html#list-of-property-types
+	// https://developer.mozilla.org/en-US/docs/Web/CSS/font-weight
+	QCOMPARE(doParse(QLatin1String("<html><head></head><body><span style=\"font-weight:100\">one</span></body></html>")), QLatin1String("<span style=\"font-weight:100\">one</span>"));
+	QCOMPARE(doParse(QLatin1String("<html><head></head><body><span style=\"font-weight:100;\">one</span></body></html>")), QLatin1String("<span style=\"font-weight:100\">one</span>"));
+	QCOMPARE(doParse(QLatin1String("<html><head></head><body><span style=\"font-weight:200;\">one</span></body></html>")), QLatin1String("<span style=\"font-weight:200\">one</span>"));
+	QCOMPARE(doParse(QLatin1String("<html><head></head><body><span style=\"font-weight:300;\">one</span></body></html>")), QLatin1String("<span style=\"font-weight:300\">one</span>"));
+	QCOMPARE(doParse(QLatin1String("<html><head></head><body><span style=\"font-weight:400;\">one</span></body></html>")), QLatin1String("<span style=\"font-weight:400\">one</span>"));
+	QCOMPARE(doParse(QLatin1String("<html><head></head><body><span style=\"font-weight:500;\">one</span></body></html>")), QLatin1String("<span style=\"font-weight:500\">one</span>"));
+	QCOMPARE(doParse(QLatin1String("<html><head></head><body><span style=\"font-weight:600\">one</span></body></html>")), QLatin1String("<b>one</b>"));
+	QCOMPARE(doParse(QLatin1String("<html><head></head><body><span style=\"font-weight:600;\">one</span></body></html>")), QLatin1String("<b>one</b>"));
+	QCOMPARE(doParse(QLatin1String("<html><head></head><body><span style=\"font-weight: 600;\">one</span></body></html>")), QLatin1String("<b>one</b>"));
+	QCOMPARE(doParse(QLatin1String("<html><head></head><body><span style=\"font-weight: 600\">one</span></body></html>")), QLatin1String("<b>one</b>"));
+	// These currently fail
```

Comment:
Not how I would expect the implementation of the tested method to work. I would expect these test cases to work (that we convert every font-weight >= 600 and "bold" to a `<b>`). Agree?